### ISSUE TITLE
[IDEA] Ability to see password in connector configuration

### ISSUE THREAD

**Issue by thorst** [nextgenhealthcare/connect#6328]
**Is your feature request related to a problem? Please describe.**
We save connection information to the configuration map so that we don't hardcode in our scripts. We store a javascript object, then in the deploy script we use JSON.parse to get it back into a js object. Then we would use it in the configuration like ${cred.un} and ${cred.pw} this works fine, but you cannot see the password so your typing blind or copying and pasting it. Just add an icon or button to view the password instead of having it be obscured.

**Comment by pacmano1** [comment]
I think you mean "option to unmask fields that are masked".      Since in your case you just want to see ${cred.pw}, not the password itself, right?  ${cred.pw} isn't evaluated until run time.

Also - I would not use $cfg for this.   Instead $cfg has _only_ db connect information.  Then everything else is sourced to $g or $gc on startup or channel deploy.   Already an object in that case.   (gonna avoid the entire topic of seeing passwords in the mirth UI, lol)

**Comment by thorst** [comment]
Correct. Currently, it's dots, and it would be nice to see the plain text.

As for $cfg, why stop at db connection info when you can store api connection info, or sftp connection info. Ultimately, we do use the channel deploy script to place the values onto the $gc. We do this so that we only have to change the connection information in 1 spot instead of dozens.

**Comment by pacmano1** [comment]
What you currently store in $cfg goes to $g:

And you store all that other information you currently store in $cfg into relevant db tables and source it on start to objects accessible to all channels (if needed in $g) or single channels via $gc.

e.g. in a global deploy script populate $g such that anywhere you can reference:
```JS
$g('this_client').api_endpoint
$g('this_client').api_username

$g('this_ehr').global_charge_item_id
```

Nothing wrong with your way of course.   I usually think in terms of abstracting this further.  e.g.  engine starts, it knows it is in test, it knows its db connect string (not mirthdb of course) is X for env test.  It goes and gets everything else from the db.
